docs/topic/billing/things.md

```diff
@@ -0,0 +1,610 @@
+
+1. There is **no charge** for any data coming in on any of the cloud providers,
+   as it is counted purely as 'data ingress'. So if a user is downloading
+   terabytes of data from the public internet, it's almost completely free.
+   Sending data out still costs money (as metered by 'data egress'), but
+   since our usage pattern is more likely to download massive data than
+   upload it, data egress doesn't amount to much in this context.
+
+2. On some cloud providers, there is a public IP fee that adds to the
+   cost of *each* node. This used to be free, but started costing money
+   only recently (starting in about 2023).
+
+   | Cloud Provider | Pricing | Link |
+   | - | - | - |
+   | GCP | $0.005 / hr for regular instances, $0.003 / hr for pre-emptible instances | [link](https://cloud.google.com/vpc/network-pricing#ipaddress) |
+   | AWS | $0.005 / hr | [link](https://aws.amazon.com/vpc/pricing/) (see 'Public IPv4 Addresses' tab)
+   | Azure | $0.004 / hr | [link](https://azure.microsoft.com/en-us/pricing/details/ip-addresses/) (Dynamic IPv4 Address for "Basic (ARM)" |
+
+This is the *default* mode of operation for our clusters, and it mostly works
+ok! We pay for each node, but do not have to worry about users racking up big
+bills because they downloaded a couple terabytes of data. Since the originating IP of each network request is the *node* the user pod is on,
+this also helps with external services (such as on-prem NASA data
+repositories) that throttle requests based on *originating IP* - all users
+on a hub are less likely to be penalized due to one user exhausting this
+quota.
+
+There are two possible negatives when using this:
+
+1. Each of our nodes is directly exposed to the public internet, which means
+   theoretically our attack surface is larger. However, these are all
+   cloud managed kubernetes nodes without any real ports open to the public,
+   so not a particularly large risk factor.
+2. Some clouds have a separate quota system for public IPs, so this may
+   limit the total size of our kubernetes cluster even if we have other
+   quotas.
+
+### Cloud NAT
+
+The alternative to having one public IP per node is to use a [Network Address Translation](https://en.wikipedia.org/wiki/Network_address_translation)
+(NAT) service. All external traffic from all nodes will go out via this
+single NAT service, which usually provides a single (or a pool of) IP addresses
+that this traffic looks to originate from. Our kubernetes nodes are not
+exposed to the public internet at all, which theoretically adds another
+layer of security as well.
+
+![Dr. Who Meme about Cloud NAT](https://hackmd.io/_uploads/H1IYevYi6.png)
+
+(one of the many memes about Cloud NAT being expensive, from [QuinnyPig](https://twitter.com/QuinnyPig/status/1357391731902341120). Many seem far more violent. See [this post](https://www.lastweekinaws.com/blog/the-aws-managed-nat-gateway-is-unpleasant-and-not-recommended/) for more information.)
```

Comment:
````suggestion
```{image} https://hackmd.io/_uploads/H1IYevYi6.png)
:alt: Dr. Who Meme about Cloud NAT

One of the many memes about Cloud NAT being expensive, from [QuinnyPig](https://twitter.com/QuinnyPig/status/1357391731902341120). Many seem far more violent. See [this post](https://www.lastweekinaws.com/blog/the-aws-managed-nat-gateway-is-unpleasant-and-not-recommended/) for more information.
```
````

Using an image directive so it's clear the text underneath is a caption